src/SpectatorView.Unity/Assets/SpectatorView/Scripts/Compositor/UnityCompositorInterface.cs

```diff
@@ -4,13 +4,68 @@
 using System;
 using System.Runtime.InteropServices;
 using System.Text;
+using UnityEngine;
 
 namespace Microsoft.MixedReality.SpectatorView
 {
-    public enum FrameProviderDeviceType : int { BlackMagic = 0, Elgato = 1, None = 2 };
+    public enum FrameProviderDeviceType : int { BlackMagic = 0, Elgato = 1, AzureKinect_DepthCamera_Off = 2, AzureKinect_DepthCamera_NFOV = 3, AzureKinect_DepthCamera_WFOV = 4, None = 5 };
+    public enum OcclusionSetting : int { RawDepthCamera = 0, BodyTracking = 1};
     public enum VideoRecordingFrameLayout : int { Composite = 0, Quad = 1 };

```

Comment:
it seems like we should have an option of none here

Reply:
or at least it seems like we may want to support enabling an azure kinect depth frame provider + a black magic rgb frame provider at some point

---
In reply to: [384742662](https://github.com/microsoft/MixedReality-SpectatorView/pull/327#discussion_r384742662) [](ancestors = 384742662)

Reply:
I think in the future if we support combining different RGB and depth camera sources, that would make sense. In the current system, you're opting into occlusion (or not) solely through the choice of a FrameProviderDeviceType.

---
In reply to: [384743198](https://github.com/microsoft/MixedReality-SpectatorView/pull/327#discussion_r384743198) [](ancestors = 384743198,384742662)